p2p/src/protocol/fees.rs

```diff
@@ -109,12 +109,10 @@ impl FeeEstimator {
         let mut fees = Vec::new();
         let snapshot = self.utxos.clone();
 
-        assert!(
-            height > self.height,
-            "Received block #{} must be higher than best block #{}",
-            height,
-            self.height,
-        );
+        // If height is less than currently best known tip height, ignore this block
+        if height < self.height {
+            return None;
+        }
```

Comment:
mh! why not use a `debug_assert`? still, preserve the logic inside the fees.rs and avoid aborting silently. 

I just say that because nakamoto use these event base action, and this type of assert are useful to keep around the code for debuggin.

Reply:
If this has to be a hard assert (debug or not), then nakamoto needs to ensure that it never asks for a block lower than the current tip height. Which is functionally infeasible.

Also an assert over this in fee calculation seems redundant to me, as its being hit in normal operation where its expected to process a block less than tip height. 

Maybe a debug log can help to notify when this happens, but anything more than that is unusable..

Reply:
Another idea is do not put blocks with height < tip height into fee estimation. Currently every block after receiving from network is subjected to fee estimation. I did not spend enough time figuring how to do that, but that should be the correct approach in this context and then the assertion can stay. 

This is the easiest fix I could make to get the tests passing for now..

Reply:
`debug_assert` works only in debug mode, and your use case looks like more of a case that can happen in production. Anyway, your consideration of the fee estimation could be an interesting idea to analyze!